net-misc/gsasl/gsasl-1.8.0.recipe

```diff
@@ -0,0 +1,69 @@
+SUMMARY="The GNU SASL tool"
+DESCRIPTION="GNU SASL is an implementation of the Simple Authentication and \
+Security Layer framework and a few common SASL mechanisms. SASL is used by \
+network servers (e.g., IMAP, SMTP) to request authentication from clients, and \
+in clients to authenticate against servers.
+
+Supported mechanisms are ANONYMOUS, EXTERNAL, LOGIN, PLAIN, SECURID, NTLM, \
+DIGEST-MD5, CRAM-MD5, SCRAM-SHA-1, SCRAM-SHA-1-PLUS, GS2-KRB5, GSSAPI."
+HOMEPAGE="https://www.gnu.org/software/gsasl/"
+COPYRIGHT="2004-2012 Simon Josefsson"
+LICENSE="GNU GPL v3
+	GNU LGPL v3"
+REVISION="1"
+SOURCE_URI="https://ftp.gnu.org/gnu/gsasl/gsasl-$portVersion.tar.gz"
+CHECKSUM_SHA256="310262d1ded082d1ceefc52d6dad265c1decae8d84e12b5947d9b1dd193191e5"
+PATCHES="gsasl-$portVersion.patchset"
+
+ARCHITECTURES="x86_gcc2 x86 x86_64"
+
+PROVIDES="
+	gsasl = $portVersion
+	cmd:gsasl = $portVersion
+	"
+REQUIRES="
+	haiku
+	lib:libgnutls
+	lib:libgsasl
+	lib:libiconv
+	lib:libintl
+	"
+
+BUILD_REQUIRES="
+	haiku_devel
+	devel:libgnutls
+	devel:libgsasl
+	devel:libiconv
+	devel:libintl
+	"
+BUILD_PREREQUIRES="
+	cmd:aclocal
+	cmd:autoconf
+	cmd:autoheader
+	cmd:automake
+#	cmd:dia
+	cmd:gcc
+	cmd:ld
+	cmd:libtoolize
+	cmd:make
+	"
+
+defineDebugInfoPackage gsasl \
+	$binDir/gsasl
+
+BUILD()
+{
+	runConfigure ./configure
+
+	# ignore lack of dia
+	make -i $jobArgs
+}
+
+INSTALL()
+{
+	# ignore lack of dia
+	make -i install
+
+	rm -rf $libDir
+	rm -rf $includeDir
+}
```

Comment:
```diff
 	rm -rf $libDir
 	rm -rf $includeDir
+	rm -rf $manDir/man3
+	rm -rf $infoDir
 }
```
I'm not sure about the infoDir. Shall we keep it? It allows to have `/system/documentation/info/gsasl.info` which should probably in be another package, libgsasl_devel, but it is not. So maybe we can keep it here until we add it to the recipe in `net-libs/libgsasl/`.
What do you think?
Any thoughts from other contributors?

BTW, I wonder if it would not be possible to drop the patch (or parts of it) by passing `CPPFLAGS="-D_BSD_SOURCE"` and `LIBS="-lbsd"` to `make`.
Would you like to investigate on this? (We can keep the PR open for a few more days if you wish, as there is no urgency.)

Reply:
About the patch, poll() is available without _BSD_SOURCE, but polling stdin strangely fails, and that forced me to change to the select() equivalent.

Reply:
OK, thanks.
If you are sure there is no way to shrink the patch then I can merge the PR soon after you add the `rm $manDir/man3`.
But if you agree, we can keep the PR open for a few days to see if someone finds a way to either avoid the patch or shrink it.

Reply:
If poll() is broken and select() works, that hints at a bug in Haiku. Please file a bug report at https://dev.haiku-os.org/ regarding that, pointing at the reproducible test case.

Reply:
@mmlr, shall I merge this PR?

Reply:
@sfanxiang, I'll let you open the bug report :)

Reply:
@fbrosson
I just figured that polling stdin is successful in this test program:
```c
#include <poll.h>
#include <stdio.h>
#include <unistd.h>

void main()
{
	struct pollfd pfd;
	pfd.fd = STDIN_FILENO;
	pfd.events = POLLIN;
	pfd.revents = 0;

	poll(&pfd, 1, -1);
	printf("%d %d\n", pfd.revents & POLLIN, pfd.revents & POLLERR);
}
```

Reply:
Well, it _does_ wait until we type something, and it _does_ not consume any character from what we type, but it _only_ quits _if/when_ we press return.
You can try. Run:
```
./stdin-poll-test ; date
```
Now type `date` but don't press return yet. First wait for e.g. 10 seconds and then press return.
You will notice that the program only exits after you press return.

I get the same results on all architectures. (x86_gcc2 primary, x86 secondary, and x86_64.)

So it looks like poll is _not_ working as expected.

Reply:
@fbrosson
Thanks for testing.

> Well, it does wait until we type something, and it does not consume any character from what we type, but it only quits if/when we press return.

This is the expected behavior. The terminal by default sends the input only when a delimiter is reached (return, EOF, ...). What I meant by "not working" is that, in gsasl (but not in the test program), polling stdin results in POLLERR.

Reply:
OK, I see. My mistake :)